homeassistant/components/shelly/light.py

```diff
@@ -256,28 +257,13 @@ def rgbw_color(self) -> tuple[int, int, int, int]:
     def color_temp(self) -> int:
         """Return the CT color value in mireds."""
         if self.control_result:
-            color_temp = self.control_result["temp"]
+            kelvin_temp = self.control_result["temp"]
         else:
-            color_temp = self.block.colorTemp
-
-        color_temp = min(self._max_kelvin, max(self._min_kelvin, color_temp))
+            kelvin_temp = self.block.colorTemp
 
-        return int(color_temperature_kelvin_to_mired(color_temp))
+        mired_temp = color_temperature_kelvin_to_mired(float(kelvin_temp))
```

Comment:
This may fail if we get an invalid value from the device (0 if I remember correctly)